kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/kotlintypes.kt

```diff
@@ -49,10 +53,41 @@ internal fun TypeName.defaultPrimitiveValue(): CodeBlock =
       FLOAT -> CodeBlock.of("0f")
       LONG -> CodeBlock.of("0L")
       DOUBLE -> CodeBlock.of("0.0")
-      UNIT, Void::class.asTypeName() -> throw IllegalStateException("Parameter with void or Unit type is illegal")
+      UNIT, Void::class.asTypeName(), NOTHING -> throw IllegalStateException("Parameter with void, Unit, or Nothing type is illegal")
       else -> CodeBlock.of("null")
     }
 
+internal fun TypeName.asTypeBlock(): CodeBlock {
+  when (this) {
+    is ParameterizedTypeName -> {
+      return if (rawType == ARRAY) {
+        CodeBlock.of("%T::class.java", copy(nullable = false))
```

Comment:
does this do the right thing? `List<String>::class.java` ?

Reply:
This is only for arrays, which are special. For other generic types it just takes the raw type.

Reply:
Resolving this since I think it was just a misunderstanding and want to build the multi-mask support on top of this, can revisit if you think the array handling is still problematic!